src/hdmf/backends/io.py

```diff
@@ -82,6 +82,14 @@ def write(self, **kwargs):
         f_builder = self.__manager.build(container, source=self.__source, root=True)
         self.write_builder(f_builder, **kwargs)
 
+        if self.external_resources_path is not None:
+            if container.get_linked_resources() is not None:
+                external_resources = container.get_linked_resources()
```

Comment:
```suggestion
            external_resources = container.get_linked_resources()
            if external_resources is not None:
```
minor refactor to get one fewer call